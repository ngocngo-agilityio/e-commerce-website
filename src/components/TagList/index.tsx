// Types
import { ITag } from '@types';

// Components
import { Tag } from '@components';

interface TagListProps {
  tags: ITag[];
  onClose: (id: number) => void;
}

const TagList = ({ tags, onClose }: TagListProps): JSX.Element => {
  return (
    <>
      {tags.map((tag) => {
        const { id, label = '' } = tag || {};

        return <Tag key={id} id={id} label={label} onClose={onClose} />;
      })}
    </>
  );
};

export default TagList;
