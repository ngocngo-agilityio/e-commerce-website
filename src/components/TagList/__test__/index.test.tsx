// Libs
import { fireEvent, render, waitFor } from '@testing-library/react';

// Mocks
import { MOCK_TAG_LIST } from '@mocks';

// Types
import { ITag } from '@types';

// Components
import TagList from '..';

const mockProps = {
  tags: MOCK_TAG_LIST,
  onClose: jest.fn(),
};

describe('Sort component', () => {
  test('should match snapshot', () => {
    const { container } = render(<TagList {...mockProps} />);

    expect(container).toMatchSnapshot();
  });

  test('should call onClose when click on the close button', () => {
    const mockTags = MOCK_TAG_LIST;
    mockTags[1] = null as unknown as ITag;

    const { getAllByTestId } = render(<TagList {...mockProps} />);

    fireEvent.click(getAllByTestId('close-btn')[0]);

    waitFor(() =>
      expect(mockProps.onClose).toHaveBeenCalledWith(MOCK_TAG_LIST[0].id),
    );
  });
});
