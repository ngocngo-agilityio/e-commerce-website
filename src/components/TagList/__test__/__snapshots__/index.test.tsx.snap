// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort component should match snapshot 1`] = `
<div>
  <span
    class="css-1ehzxp"
  >
    <span
      class="css-zvlevn"
    >
      Women
    </span>
    <button
      aria-label="close"
      class="css-sorhh"
      data-testid="close-btn"
      type="button"
    >
      <svg
        class="chakra-icon css-1md6f3y"
        focusable="false"
        viewBox="0 0 512 512"
      >
        <path
          d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
          fill="currentColor"
        />
      </svg>
    </button>
  </span>
  <span
    class="css-1ehzxp"
  >
    <span
      class="css-zvlevn"
    >
      Man
    </span>
    <button
      aria-label="close"
      class="css-sorhh"
      data-testid="close-btn"
      type="button"
    >
      <svg
        class="chakra-icon css-1md6f3y"
        focusable="false"
        viewBox="0 0 512 512"
      >
        <path
          d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
          fill="currentColor"
        />
      </svg>
    </button>
  </span>
</div>
`;
